src/Language/C/Data/Ident.hs

```diff
@@ -47,29 +49,32 @@ isAnonymousRef :: SUERef -> Bool
 isAnonymousRef (AnonymousRef _) = True
 isAnonymousRef _ = False
 
+type Ident = Identifier NodeInfo
+
 -- | C identifiers
-data Ident = Ident String       -- lexeme
+data Identifier a = Ident String       -- lexeme
                    {-# UNPACK #-}   !Int     -- hash to speed up equality check
-                   NodeInfo                   -- attributes of this ident. incl. position
-             deriving (Data,Typeable,Show, Generic) -- Read
+                   a                   -- attributes of this ident. incl. position
+             deriving (Data,Typeable,Show,Functor,Generic, Generic1) -- Read
 
-instance NFData Ident
+instance NFData a => NFData (Identifier a)
 
 -- the definition of the equality allows identifiers to be equal that are
 -- defined at different source text positions, and aims at speeding up the
 -- equality test, by comparing the lexemes only if the two numbers are equal
 --
```

Comment:
I wonder if being polymorphic over the NodeInfo it would be worth removing the weirdy Eq/Ord instance. One could always do something like ``(==) `on` void`` to regain the old behavior.

Reply:
Yes! great catch! I think it would make sense to stay with the stock implementations.